feat: added touch/slide on prematch option buttons (#1529)

Mandolinx opened the pull request:
This fixes issue: #1529

- created a function called buttonSlide in button.ts for pre-match button game options to touch and slide
- created a new html class in index.ejs called "dragIt"
- made all radio labels have class "dragIt"
- added the buttonSlide function to script.ts

clicking on a pre-match button game option and dragging to the next button will click that next button and reset original button.

DreadKnight commented on src/index.ejs:
```diff
@@ -74,91 +74,121 @@
 									<div id="playerMode" class="typeRadio">
 										<input type="radio" id="p2" name="playerMode" value="2" checked>
-										<label for="p2">1 versus 1</label>
+										<label for ="p2" class="draggable-label">1 versus 1</label>
```
Space, the final frontier.

DreadKnight commented on src/index.ejs:
```diff
@@ -74,91 +74,121 @@
+										<label for ="p2" class="draggable-label">1 versus 1</label>
 										<input type="radio" id="p4" name="playerMode" value="4">
-										<label for="p4">2 versus 2</label>
+										<label for="p4" class="draggable-label">2 versus 2</label>
```
Maybe class name should be way shorter, like "dragit" or so.

DreadKnight commented on src/index.ejs:
```diff
@@ -74,91 +74,121 @@
 									</div>
 								</div>
+								<!--Script to add sliding to prematch option buttons-->
+								<script>
```
Best to make index file smaller (as we even plan to split it into files). Would be best if script was separate file in `src/ui` and be imported using webpack or whatever.

DreadKnight commented on src/index.ejs:
```diff
@@ -74,91 +74,121 @@
+										  selectedRadio = radio.previousElementSibling;
+										});
+										
+										// check new dragged radio/button
```
Comment style consistency: capitalized first letter, `// Check...`

DreadKnight reviewed (changes requested):
Heya! Some things that should probably be poked.

DreadKnight commented:
@Mandolinx  I've tested this and works well, however, I've opened a review regarding the coding itself to polish things a bit.

Mandolinx commented:
Hi @DreadKnight I've just updated my code such that the script is now in a separate file in UI called "slider.js", and fixed various syntax and style issues. During my process, I was not familiar with webpack so I'm not entirely sure if my method of importing is suitable for this issue so that may be an area of focus in the review. Also, my script file that does the slide is in a separate but it only contains a function, so I was wondering whether it'd be better to just add the function directly to script.ts or add the function to Button.ts. Thanks! 

p.s. interface.js is said to be changed but all I did was copy and paced the interface file from repo because git pull was not working.

DreadKnight commented:
> Hi @DreadKnight I've just updated my code such that the script is now in a separate file in UI called "slider.js", and fixed various syntax and style issues. During my process, I was not familiar with webpack so I'm not entirely sure if my method of importing is suitable for this issue so that may be an area of focus in the review. Also, my script file that does the slide is in a separate file but it only contains a function, so I was wondering whether it'd be better to just add the function directly to script.ts or add the function to Button.ts. Thanks!
> 
> p.s. interface.js is said to be changed but all I did was copy and paced the interface file from repo because git pull was not working.

Moving the stuff in `script.ts` it doesn't make sense imo, because that's an initial set-up thingy, but it might make sense in `button.ts` if the file is about the pre-match screen buttons (wish it had some description), so feel free to move if so and also remove "Used for prematch button game options to slide" comment line if moving it there.

Regarding `interface.js`, you usually don't need to pull master to latest when it comes to PRs; you haven't touched that file in your commits and the copy-pasting that you did actually reverts a patch of mine that I might have missed. Just exclude your changes by reverting file to what it was ideally.

Mandolinx commented:
> Moving the stuff in `script.ts` it doesn't make sense imo, because that's an initial set-up thingy, but it might make sense in `button.ts` if the file is about the pre-match screen buttons (wish it had some description), so feel free to move if so and also remove "Used for prematch button game options to slide" comment line if moving it there.
> 
> Regarding `interface.js`, you usually don't need to pull master to latest when it comes to PRs; you haven't touched that file in your commits and the copy-pasting that you did actually reverts a patch of mine that I might have missed. Just exclude your changes by reverting file to what it was ideally.

As requested, I reverted to my original version of interface and I put the buttonSlider into the button file. I also changed the function to use jquery since I was having problems earlier copy and pasting my old function into the file without it, but it has the same functionality so everything should be fine for that. Thanks!

DreadKnight commented:
> > Moving the stuff in `script.ts` it doesn't make sense imo, because that's an initial set-up thingy, but it might make sense in `button.ts` if the file is about the pre-match screen buttons (wish it had some description), so feel free to move if so and also remove "Used for prematch button game options to slide" comment line if moving it there.
> > Regarding `interface.js`, you usually don't need to pull master to latest when it comes to PRs; you haven't touched that file in your commits and the copy-pasting that you did actually reverts a patch of mine that I might have missed. Just exclude your changes by reverting file to what it was ideally.
> 
> @DreadKnight As requested, I reverted to my original version of interface and I put the buttonSlider into the button file. I also changed the function to use jquery since I was having problems earlier copy and pasting my old function into the file without it, but it has the same functionality so everything should be fine for that. Thanks!

@Mandolinx Thanks way better, good job! 👍🏻
